src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -667,6 +668,44 @@ public override ImmutableArray<string> TupleElementNames
         }
 
         /// <summary>
+        /// Get the fields for the tuple's elements (in order and cached).
+        /// </summary>
+        public override ImmutableArray<FieldSymbol> TupleElementFields
+        {
+            get
+            {
+                if (_lazyFields.IsDefault)
+                {
+                    ImmutableInterlocked.InterlockedInitialize(ref _lazyFields, CollectTupleElementFields());
+                }
+
+                return _lazyFields;
+            }
+        }
+
+        private ImmutableArray<FieldSymbol> CollectTupleElementFields()
+        {
+            var builder = ArrayBuilder<FieldSymbol>.GetInstance(_elementTypes.Length);
+            builder.SetItem(_elementTypes.Length - 1, null);
```

Comment:
This is subtle, I think you can simply pass null as the second parameter for ArrayBuilder<FieldSymbol>.GetInstance to have the builder allocate the items.